cli/azd/pkg/experimentation/assignment.go

```diff
@@ -0,0 +1,186 @@
+package experimentation
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/azure/azure-dev/cli/azd/internal/tracing/resource"
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+)
+
+// cCacheFileNamePattern is the pattern used to generate the cache file name. It is formatted
+// using the machine ID.
+const cCacheFileNamePattern = "mid-%s.cache"
```

Comment:
Do wonder if `assign` should be part of the file name just in case we ever use the directory to store other cache files.

Reply:
If we adopt the new format I outlined, I will change this to just `assign.cache` (since we will only ever have one, as a single file contains all the entries).

Reply:
Based on the discussion above, this is now just called `assign.cache` and contains a single entry and an expiration time.